app/src/sharedTest/java/org/oppia/app/topic/train/TopicTrainFragmentTest.kt

```diff
@@ -2,36 +2,139 @@ package org.oppia.app.topic.train
+    TopicActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    skillIdList.add("test_skill_id_0")
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_displaySkills_startButtonIsInactive() {
+    ActivityScenario.launch(TopicActivity::class.java).use {
+      onView(withId(R.id.master_skills_text_view)).check(matches(withText(R.string.topic_train_master_these_skills)))
+      onView(atPosition(R.id.skill_recycler_view,0)).check(matches(hasDescendant(withId(R.id.skill_check_box))))
+      onView(withId(R.id.topic_train_start_button)).check(matches(not(isClickable())))
+    }
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_isSuccessful() {
+    ActivityScenario.launch(TopicActivity::class.java).use {
+      onView(atPosition(R.id.skill_recycler_view,0)).perform(click())
+      onView(atPosition(R.id.skill_recycler_view,1)).perform(click())
+    }
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_startButtonIsActive() {
+    ActivityScenario.launch(TopicActivity::class.java).use {
+      onView(atPosition(R.id.skill_recycler_view,0)).perform(click())
+      onView(withId(R.id.topic_train_start_button)).check(matches(isClickable()))
+    }
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_deselectSkills_isSuccessful() {
+    ActivityScenario.launch(TopicActivity::class.java).use {
+      onView(atPosition(R.id.skill_recycler_view,0)).perform(click())
+      onView(atPosition(R.id.skill_recycler_view,0)).perform(click())
+    }
+  }
+
   @Test
-  fun testTopicTrainFragment_loadFragment_textIsDisplayed() {
+  fun testTopicTrainFragment_loadFragment_selectSkills_deselectSkills_startButtonIsInactive() {
     ActivityScenario.launch(TopicActivity::class.java).use {
-      onView(withId(R.id.dummy_text_view)).check(matches(withText("This is dummy TextView for testing")))
+      onView(atPosition(R.id.skill_recycler_view,0)).perform(click())
+      onView(atPosition(R.id.skill_recycler_view,0)).perform(click())
+      onView(withId(R.id.topic_train_start_button)).check(matches(not(isClickable())))
     }
   }
 
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_clickStartButton_skillListTransferSuccessfully() {
+    activityTestRule.launchActivity(null)
+    onView(atPosition(R.id.skill_recycler_view,0)).perform(click())
+    onView(withId(R.id.topic_train_start_button)).perform(click())
+    intended(hasComponent(QuestionPlayerActivity::class.java.name))
+    intended(hasExtra(QuestionPlayerActivity.getIntentKey(), skillIdList))
+  }
+
+  @Test
+  fun testTopicTrainFragment_loadFragment_selectSkills_configurationChange_skillsAreSelected() {
+    activityTestRule.launchActivity(null)
+    onView(atPosition(R.id.skill_recycler_view,0)).perform(click())
+    activityTestRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
```

Comment:
I don't believe this is sufficient for a configuration change. Suggest instead: ``ActivityScenario.recreate()``

Ditto elsewhere for configuration changes.

Reply:
This means I have add this or do I have to remove the current line and replace this one?

Reply:
Using both and merge code for now, because the test-cases passes in all possible combinations, using current code only, using your code only and using both.